### PR TITLE
feat: badge cleanup

### PR DESCRIPTION
**Pull request by seedy**
### Description

Cleaning up badge component, NB: this applies no design system changes. 

This will be done, hopefully in a future step.

- added missing compoundVariants programatically
- added all color variants in story programatically
  - default color variant (`gray`) is duplicated to make sure it's clear to the user
- cleaned up theme file

### Package changes

- updated `rollup-plugin-typescript2` from `^0.29.0` to `^0.30.0` due to a build issue
  - CI: https://github.com/traefik/faency/runs/4433331267?check_suite_focus=true
  - https://github.com/alexjoverm/typescript-library-starter/issues/310#issuecomment-811451090

### CODE REVIEW

File: components/Badge/Badge.tsx [hunk 1]
```diff
@@ -63,106 +87,14 @@ export const Badge = styled('span', {
...
-      interactive: true,
-      variant: 'neon',
-      css: { '&:focus-visible': { boxShadow: '0 0 0 1px $colors$neon9' } },
-    },
-    {
-      interactive: true,
-      variant: 'orange',
-      css: { '&:focus-visible': { boxShadow: '0 0 0 1px $colors$orange9' } },
-    },
-    {
```
Comment (seedy) [inline components/Badge/Badge.tsx:131]: :information_source: missing compoundVariant purple

File: components/Badge/Badge.stories.tsx [hunk 2]
```diff
@@ -78,21 +66,9 @@ export const Interactive: ComponentStory<typeof BadgeForStory> = (args) => (
...
-    </Badge>
-    <Badge as="button" {...args} variant="blue">
-      Blue
-    </Badge>
-    <Badge as="button" {...args} variant="neon">
-      Neon
-    </Badge>
-    <Badge as="button" {...args} variant="orange">
-      Orange
-    </Badge>
```
Comment (seedy) [inline components/Badge/Badge.stories.tsx:95]: :information_source: missing purple in story

File: components/Badge/Badge.stories.tsx [hunk 3]
```diff
@@ -23,21 +23,9 @@ export default {
...
-    <Badge {...args} variant="blue">
-      Blue
-    </Badge>
-    <Badge {...args} variant="neon">
-      Neon
-    </Badge>
-    <Badge {...args} variant="orange">
-      Orange
-    </Badge>
```
Comment (seedy) [inline components/Badge/Badge.stories.tsx:40]: :information_source: missing purple in story

File: package.json [hunk 4]
```diff
@@ -69,7 +69,7 @@
     "react": "17.0.1",
     "react-dom": "17.0.1",
     "rollup": "^2.33.1",
-    "rollup-plugin-typescript2": "^0.29.0",
+    "rollup-plugin-typescript2": "^0.30.0",
```
Comment (seedy) [inline package.json:72]: :information_source: Check PR description

**Review by matthieuh — APPROVED** [review]
LGTM

**Review by paulocfjunior — APPROVED** [review]
LGTM

**Pull request merged**